packages/serverless-workflow-editor/src/editor/ServerlessWorkflowEditor.tsx

```diff
@@ -131,24 +131,27 @@ const RefForwardingServerlessWorkflowEditor: React.ForwardRefRenderFunction<
     []
   );
 
-  const setValidationErrors = (errors: editor.IMarker[]) => {
-    if (!initialContent) {
-      return;
-    }
-    const notifications: Notification[] = errors.map((error: editor.IMarker) => ({
-      type: "PROBLEM",
-      path: initialContent.path,
-      severity: "ERROR",
-      message: `${error.message}`,
-      position: {
-        startLineNumber: error.startLineNumber,
-        startColumn: error.startColumn,
-        endLineNumber: error.endLineNumber,
-        endColumn: error.endColumn,
-      },
-    }));
-    props.setNotifications(initialContent.path, notifications);
-  };
+  const setValidationErrors = useCallback(
+    (errors: editor.IMarker[]) => {
+      if (!initialContent) {
+        return;
+      }
+      const notifications: Notification[] = errors.map((error: editor.IMarker) => ({
+        type: "PROBLEM",
+        path: initialContent.path,
+        severity: "ERROR",
+        message: `${error.message}`,
+        position: {
+          startLineNumber: error.startLineNumber,
+          startColumn: error.startColumn,
+          endLineNumber: error.endLineNumber,
+          endColumn: error.endColumn,
+        },
+      }));
+      props.setNotifications(initialContent.path, notifications);
+    },
+    [initialContent, props]
```

Comment:
```suggestion
      props.setNotifications.apply(undefined, initialContent.path, notifications);
    },
    [initialContent, props.setNotifications]
```

Read more about it here: https://github.com/facebook/react/issues/16265

There are other solutions too, if you want to explore options.